sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/AvroUtils.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.gcp.bigquery;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.beam.sdk.schemas.Schema.Field;
+import org.apache.beam.sdk.schemas.Schema.TypeName;
+import org.joda.time.Instant;
+
+/** Utils to help convert Apache Avro types to Beam types. */
+public class AvroUtils {
+  public static Object convertAvroFormat(Field beamField, Object value) throws RuntimeException {
```

Comment:
so we don't support recursive rows or array types?

Reply:
Yes because I didn't find a nested structure from Schema.Field. In order to convert data in Arvo format, beam type (INT32, etc.) has to be known.  However, Schema.Field does not have a recursive structure to get type information.

Reply:
I'm not sure I understand - Schema.Field can be of type ROW, in which case the field has a nested schema.

Reply:
Can we know the `STRING` from  `ROW<ROW<ROW<STRING>>` from Schema.Field?

Reply:
O actually you are right, I missed this:
```
    // For ROW types, returns the schema for the row.
    @Nullable
    public abstract Schema getRowSchema();
```

I do can get nested schema/fields.